src/foundry/common/data/data.mjs/tokenData.d.ts

```diff
@@ -0,0 +1,540 @@
+  effects: DocumentField<string[]> & {
+    type: String[];
+    required: true;
+    default: string[];
+  };
+  overlayEffect: typeof fields.STRING_FIELD;
+  alpha: typeof fields.ALPHA_FIELD;
+  hidden: typeof fields.BOOLEAN_FIELD;
+  vision: DocumentField<boolean> & {
+    type: Boolean;
+    required: true;
+    default: (data: { readonly dimSight?: number; readonly brightSight?: number }) => boolean;
+  };
+  dimSight: FieldReturnType<typeof fields.REQUIRED_NUMBER, VisionFieldOptions>;
+  brightSight: FieldReturnType<typeof fields.REQUIRED_NUMBER, VisionFieldOptions>;
+  dimLight: FieldReturnType<typeof fields.REQUIRED_NUMBER, VisionFieldOptions>;
+  brightLight: FieldReturnType<typeof fields.REQUIRED_NUMBER, VisionFieldOptions>;
+  sightAngle: typeof fields.ANGLE_FIELD;
+  lightAngle: typeof fields.ANGLE_FIELD;
+  lightColor: typeof fields.COLOR_FIELD;
+  lightAlpha: FieldReturnType<typeof fields.ALPHA_FIELD, { default: 0.25 }>;
+  lightAnimation: DocumentField<AnimationData> & {
+    type: AnimationData;
+    required: true;
+    default: {};
+  };
+  disposition: DocumentField<CONST.TokenDisposition> & {
+    type: Number;
+    required: true;
+    default: typeof CONST.TOKEN_DISPOSITIONS.HOSTILE;
+    validate: (n: any) => boolean;
+    validationError: 'Invalid {name} {field} which must be a value in CONST.TOKEN_DISPOSITIONS';
+  };
+  displayBars: DocumentField<CONST.TokenDisplayMode> & {
+    type: Number;
+    required: true;
+    default: typeof CONST.TOKEN_DISPLAY_MODES.NONE;
+    validate: (m: any) => boolean;
+    validationError: 'Invalid {name} {field} which must be a value in CONST.TOKEN_DISPLAY_MODES';
+  };
+  bar1: DocumentField<TokenBarData> & {
+    type: TokenBarData;
+    required: true;
+    default: () => { attribute: typeof game['system']['data']['primaryTokenAttribute'] | null };
+  };
+  bar2: DocumentField<TokenBarData> & {
+    type: TokenBarData;
+    required: true;
+    default: () => { attribute: typeof game['system']['data']['secondaryTokenAttribute'] | null };
+  };
+  flags: typeof fields.OBJECT_FIELD;
+}
+
+export interface TokenDataProperties {
+  /**
+   * The Token _id which uniquely identifies it within its parent Scene
+   * @defaultValue `null`
+   */
+  _id: string | null;
+
+  /**
+   * The name used to describe the Token
+   */
+  name?: string;
+
+  /**
+   * The display mode of the Token nameplate, from CONST.TOKEN_DISPLAY_MODES
+   * @defaultValue `CONST.TOKEN_DISPLAY_MODES.NONE`
+   */
+  displayName: CONST.TokenDisplayMode;
+
+  /**
+   * The _id of an Actor document which this Token represents
+   * @defaultValue `null`
+   */
+  actorId: string | null;
+
+  /**
+   * Does this Token uniquely represent a singular Actor, or is it one of many?
+   * @defaultValue `false`
+   */
+  actorLink: boolean;
+
+  /**
+   * Token-level data which overrides the base data of the associated Actor
+   * @defaultValue `{}`
+   */
+  actorData: ActorDataProperties['data'];
```

Comment:
Shouldn't this be a `Partial` / `DeepPartial` then...? Maybe check what foundry actually does at runtime